tests/unittest_qmail-send.c

```diff
@@ -0,0 +1,239 @@
+#define DEPRECATED_FUNCTIONS_REMOVED
+#include <check.h>
+#include "qmail-send-without-main.c"
+
+extern struct job *jo;
+extern int numjobs;
+
+// START test helpers for the API we might prefer from qmail-send
+
+#define available 0
+#define unavailable 1
+
+static void job_init_this_many(int numslots) {
+  numjobs = numslots;
+  job_init();
+}
+
+static void occupy_job(unsigned int index) {
+  jo[index].refs = unavailable;
+}
+
+static void occupy_all_jobs(void) {
+  for (int j = 0; j < numjobs; j++) {
+    occupy_job(j);
+  }
+}
+
+static void release_job(unsigned int index) {
+  jo[index].refs = available;
+}
+
+// END test helpers for the API we might prefer from qmail-send
+
+// when job_avail() returns an index, switch to ck_assert_int_eq
+#define assert_job_avail ck_assert_uint_eq
+
+static void assert_job_at_index(unsigned int index, int availability) {
+  ck_assert_int_eq(jo[index].refs, availability);
+}
+
+static void setup(void) {
+  numjobs = -77;
+}
+
+static void teardown(void) {
+  numjobs = -88;
+}
+
+START_TEST(test_qmail_job_init_sets_expected_default_state)
+{
+  ck_assert_ptr_null(jo);
+
+  job_init_this_many(4);
+
+  ck_assert_ptr_nonnull(jo);
+  ck_assert_uint_eq(numjobs, 4);
+  assert_job_at_index(0, available);
```

Comment:
You set 4 jobs, but only check 2. Please check them all in a loop, and maybe also check that all other slots are unused.

Please always check all slots in all tests. This is unit tests, a bit of paranoia can't hurt here.

Reply:
This makes, sense, and I do believe we originally were looping, but decided against it later on. @schmonz  maybe you can help recall why we retracted from that idea. I don't think there's anything wrong with putting `line 57` in a loop.